File: problem58/problem58.js
/**
 * Problem 58
 */

var IntegerArithmetic = require('../libraries/IntegerArithmetic.js');
var currentVal = 1; var nbPrimes = 0;
var count = 1; var ratio = 1;
var i = 2;
while(true) {
	for (var j = 0; j < 4; j++) {
		currentVal += i;
		count++;
		if (IntegerArithmetic.isPrime(currentVal)) nbPrimes++;
		if (j == 3) ratio = nbPrimes / count;
	}
	if (ratio < 0.1) {
		console.log('The solution is ' + (i + 1));
		break;
	}
	i += 2;
}

File: problem58/Problem58.java
import java.util.*;
/**
 * Problem 58
 */

public class Problem58 {
	public static void main(String[] arg) {
		long currentVal = 1; int nbPrimes = 0;
		int count = 1; float ratio = 1;
		int i = 2;
		while (true) {
			for (int j = 0; j < 4; j++) {
				currentVal += i;
				count++;
				if (IntegerArithmetic.isPrime(currentVal)) nbPrimes++;
				if (j == 3) ratio = (float)nbPrimes / (float)count;
			}
			if (ratio < 0.1) {
				System.out.println("The solution is " + (i + 1));
				break;
			} 
			i += 2;
		}
	}
}